src/legacy/core_plugins/kibana/public/management/sections/index_patterns/edit_index_pattern/create_edit_field/create_edit_field.tsx

```diff
@@ -0,0 +1,111 @@
+/*
+ * Licensed to Elasticsearch B.V. under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch B.V. licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+import React from 'react';
+import { withRouter, RouteComponentProps } from 'react-router-dom';
+// @ts-ignore
+import { FieldEditor } from 'ui/field_editor';
+
+import { EuiFlexGroup, EuiFlexItem } from '@elastic/eui';
+import { i18n } from '@kbn/i18n';
+import { IndexHeader } from '../index_header';
+import { IndexPattern, IndexPatternField } from '../../../../../../../../../plugins/data/public';
+import { ChromeDocTitle, NotificationsStart } from '../../../../../../../../../core/public';
+
+type CreateEditFieldProps = RouteComponentProps & {
```

Comment:
Please let's use interface here. You can extend it from RouteComponentProps

Reply:
done